### PR TITLE
chore: improve nox maintenance path

### PR DESCRIPTION
**Pull request by lphuc2250gma**
Summary:
- Add focused unit tests covering edge cases in nox/tox_to_nox.py (e.g., empty envlist, malformed factors, deps with extras, commands that span multiple lines) and tighten/complete type annotations on its helper functions so the module passes stricter mypy checks.
- Keep the change narrow so it is straightforward to review.

Notes:
- I kept this scoped to the relevant implementation and tests.

### CODE REVIEW

File: nox/tox_to_nox.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def __dir__() -> list[str]:
 )
 
 
-def wrapjoin(seq: Iterable[Any]) -> str:
+def wrapjoin(seq: Iterable[str]) -> str:
```
Comment (henryiii) [inline nox/tox_to_nox.py:51]: Not correct, this takes anything that can be converted to a string. And in Python, everything can because `__str__` is defined on `object`.

File: nox/tox_to_nox.py [hunk 2]
```diff
@@ -87,7 +87,7 @@ def main() -> None:
 
         config[name] = dict(section)
         # Convert set_env from string to dict
-        set_env = {}
+        set_env: dict[str, str] = {}
```
Comment (henryiii) [inline nox/tox_to_nox.py:90]: Why? Type checkers can infer it below. Unless a specific type checker can't? Don't we run mypy?
